libuavcan/src/transport/uc_transfer_sender.cpp

```diff
@@ -134,6 +140,17 @@ int TransferSender::send(const uint8_t* payload, unsigned payload_len, Monotonic
             UAVCAN_ASSERT(offset <= int(payload_len));
             if (offset >= int(payload_len))
             {
+                for(unsigned i = 0; i < padding_bytes; ++i)
+                {
+                    if(0 == frame.pushBackPayload(BytePaddingPattern))
+                    {
+                        UAVCAN_TRACE("TransferSender", "Frame payload padding failure");
+                        registerError();
+                        // The calculations above should make it impossible
+                        // to get here.
```

Comment:
These kinds of errors should be accompanied with `UAVCAN_ASSERT(false);` in order to amplify errors in a development environment.

Reply:
Replacing `UAVCAN_TRACE` with `UAVCAN_ASSERT(false)`